mitodl/open-discussions
# UI for Staff Lists Listing page

#### Pre-Flight checklist


#### What are the relevant tickets?
#3959 

#### How should this be manually tested?
1. Visit http://od.odl.local:8063/infinite/stafflists as an authenticated user. *Unauthenticated access to this page should be handled as part of https://github.com/mitodl/open-discussions/issues/3864*
2. Open network tab in dev tools
3. Create some staff lists via "Create List" button+dialog (Learning lists and Learning paths).
    - verify the POST requests are sent to `stafflists/` (not `userlists/`)
4. There's no UI for adding staff list items yet. At `http://od.odl.local:8063/api/v0/stafflists/LIST_ID/items`, add some new items to your list. 
    - These weren't created via UI, so UI will not update automatically.
    - **After a refresh:** check that the stafflists/ page displays correct number of items.
5. Try editing stafflists via "..." -> "Edit". After saving, re-open the edit dialog and you should see your changes there.
6. Try deleting stafflists. 
7. Try creating/editing/deleting userlists at  http://od.odl.local:8063/infinite/lists ... Check the API requests go to `userlists/` not `stafflists/`

#### Screenshots (if appropriate)
<img width="1073" alt="Screenshot 2023-05-01 at 11 51 44 AM" src="https://user-images.githubusercontent.com/9010790/235481845-579d3e94-9f6b-4796-8b2c-58106829b421.png">

<img width="1054" alt="Screenshot 2023-05-01 at 11 51 59 AM" src="https://user-images.githubusercontent.com/9010790/235481893-e932f0f7-58d1-4e3f-9047-36ffa5937209.png">
